app/assets/javascripts/tables/base/editable_table.js

```diff
@@ -34,9 +34,17 @@ var EditableTable = (function () {
     function extractTextfromCells(row) {
         return $(row).find("td.editable").toArray().map(function (cell) {
             var input = $(cell).find("select:visible").length > 0 ? 'select' : 'input',
-                elem  = $(cell).find(input);
+                elem  = $(cell).find(input),
+                type  = elem.data('type'),
+                value = $.trim(elem.rawValue());
+
+            if (type === 'float') {
+                value = parseFloat(value);
+            } else if (type == 'integer') {
+                value = parseInt(value);
+            }
```

Comment:
> Is it perhaps worth making these names consistent?

Yes. I am not sure which way or the other seems 'better'. I am not a big fan of the `number_field_tag` mainly due to it's rather crappy implementation in Firefox 😆 I'm thinking to use the `float` and `integer` fields on the topology form aswell for the sake of consistency. I'll also take a look at the `profile_matrix` because I think I skipped that one.